great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -329,16 +335,14 @@ def run(
             rules=rules, reconciliation_directives=reconciliation_directives
         )
 
-        if expectation_suite_name is None:
-            expectation_suite_name = (
-                f"tmp.profiler_{self.__class__.__name__}_suite_{str(uuid.uuid4())[:8]}"
+        if expectation_suite is None:
```

Comment:
I think we can safely `and` these conditions; it should improve readability

Reply:
The possible inputs here are as follows:
```
suite + suite_name       -> AssertionError
suite + no suite_name    -> Skip this whole conditional
no suite + suite_name    -> Create a suite with the given name
no suite + no suite_name -> Create a suite with a generated name
```

In that second case, I think we require this nested set of conditionals right? I don't have a suite but I do have a name - using `and` would skip the logic to create a suite and leave us in a situation where `expectation_suite` is `None`.

Reply:
@cdkini @donaldheppner A similar logic should already exist in `Validator` -- so can be adapted herein as well.

Reply:
oh yeah, I see it now